### PR TITLE
[FEATURE] added supplier to form

### PR DESCRIPTION
**Pull request by michaelzangerle**
__Tasks:__
- [x] done
- [ ] got feedback

__Informations:__

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Tests pass?   | yes
| Fixed tickets | none
| Doc           | none

__Requires:__
- https://github.com/sulu-cmf/sulu/pull/891

### CODE REVIEW

File: Api/Product.php [hunk 1]
```diff
@@ -305,6 +305,25 @@ public function setSupplier($supplier)
...
+     * @VirtualProperty
+     * @SerializedName("supplier")
+     */
+    public function getSupplier()
+    {
+        $values = null;
+        $supplier = $this->entity->getSupplier();
+        if ($supplier !== null) {
+            // Returns no api entity because it will cause a nesting level exception
+            $values = array();
```
Comment (christianbader) [inline Api/Product.php]: use either array(key => value);
or remove
values = array();

File: Product/ProductManager.php [hunk 2]
```diff
@@ -823,6 +834,24 @@ public function save(
...
+
+        if (isset($data['supplier']) && isset($data['supplier']['id'])) {
+            $supplierId = $data['supplier']['id'];
+            /** @var Account $supplierId */
+            $supplier = $this->accountRepository->find($supplierId);
+            if (!$supplier) {
+                throw new ProductDependencyNotFoundException(self::$accountsSupplierEntityName, $supplierId);
+            }
+            $product->setSupplier($supplier);
+        } else if (isset($data['supplier']) && (!isset($data['supplier']['id']) || $data['supplier']['id'] === null)){
```
Comment (christianbader) [inline Product/ProductManager.php]: isset does return null if $data['supplier']['id'] == null

**Comment by christianbader** [comment]
tests? else ack.

**Comment by michaelzangerle** [comment]
Tests are working!

**Pull request merged**